tests/serialization/test_dataframe.py

```diff
@@ -530,6 +530,81 @@ def test_predicate_pushdown_null_col(
     )
 
 
+@pytest.mark.parametrize(
+    "df, op, value, expected_index",
+    [
+        (
+            pd.DataFrame({"u": pd.Series([None, "x", np.nan], dtype=object)}),
+            "==",
+            None,
+            [0, 2],
+        ),
+        (
+            pd.DataFrame({"u": pd.Series([None, "x", np.nan], dtype=object)}),
+            "in",
+            [None],
+            [0, 2],
+        ),
+        (
+            pd.DataFrame({"u": pd.Series([None, "x", np.nan], dtype=object)}),
+            "!=",
+            None,
+            [1],
+        ),
+        (
+            pd.DataFrame({"u": pd.Series([None, "x", np.nan], dtype=object)}),
+            "in",
+            [None, "x"],
+            [0, 1, 2],
+        ),
+        (
+            pd.DataFrame({"f": pd.Series([np.nan, 1.0, np.nan], dtype=float)}),
+            "==",
+            np.nan,
+            [0, 2],
+        ),
+        (
+            pd.DataFrame({"f": pd.Series([np.nan, 1.0, np.nan], dtype=float)}),
+            "in",
+            [np.nan],
+            [0, 2],
+        ),
+        (
+            pd.DataFrame({"f": pd.Series([np.nan, 1.0, np.nan], dtype=float)}),
+            "!=",
+            np.nan,
+            [1],
+        ),
+        (
+            pd.DataFrame({"f": pd.Series([np.nan, 1.0, np.nan], dtype=float)}),
+            "in",
+            [np.nan, 1.0],
+            [0, 1, 2],
+        ),
+    ],
+)
+@predicate_serialisers
+@pytest.mark.parametrize("predicate_pushdown_to_io", [True, False])
+def test_predicate_parsing_null_values(
+    store, df, op, value, expected_index, predicate_pushdown_to_io, serialiser
+):
+    key = serialiser.store(store, "prefix", df)
+
+    expected = df.iloc[expected_index].copy()
+    predicates = [[(df.columns[0], op, value)]]
+    result = serialiser.restore_dataframe(
+        store,
+        key,
+        predicate_pushdown_to_io=predicate_pushdown_to_io,
+        predicates=predicates,
+    )
+    pdt.assert_frame_equal(
+        result.reset_index(drop=True),
+        expected.reset_index(drop=True),
+        check_dtype=serialiser.type_stable,
+    )
+
```

Comment:
I'm positively surprised to see this changed. I was working under the assumption that the null_count is not reliable (I'm pretty sure this wasn't back in earlier pyarrow versions). Just to satisfy my paranoia, could you please add the following test to verify our assumptions about the null count statistics? (Feel free to adapt if you see improvements)

```suggestion

def test_null_count(store):
    serialiser = ParquetSerializer(chunk_size=2)
    df = pd.DataFrame(
        {
            "partial_nulls_int": [1, 2, 3, None],
            "partial_nulls_float": [1.0, 2.2, 3.3, np.nan],
            "partial_nulls_obj": [1.0, 2.2, 3.3, np.nan],
            "partial_nulls_obj_mixed": [1.0, 2.2, None, np.nan],
            "nulls_reverse_rg": [3.3, np.nan, 1.0, 2.0],
        }
    )
    key = serialiser.store(store, "prefix", df)

    reader = pa.BufferReader(store.get(key))
    parquet_file = ParquetFile(reader)

    assert parquet_file.num_row_groups == 2

    rg_0 = parquet_file.metadata.row_group(0)
    for ix in range(0, 4):
        assert rg_0.column(ix).statistics.null_count == 0
    assert rg_0.column(4).statistics.null_count == 1

    rg_1 = parquet_file.metadata.row_group(1)
    for ix in range(0, 3):
        assert rg_1.column(ix).statistics.null_count == 1
    assert rg_1.column(3).statistics.null_count == 2
    assert rg_1.column(4).statistics.null_count == 0
```